libraries/AP_InertialSensor/AP_InertialSensor_config.h

```diff
@@ -48,7 +48,7 @@
 #endif
 
 #ifndef AP_INERTIALSENSOR_HARMONICNOTCH_ENABLED
-#define AP_INERTIALSENSOR_HARMONICNOTCH_ENABLED AP_INERTIALSENSOR_ENABLED
+#define AP_INERTIALSENSOR_HARMONICNOTCH_ENABLED (AP_INERTIALSENSOR_ENABLED && !defined(HAL_BUILD_AP_PERIPH))
```

Comment:
set ` AP_INERTIALSENSOR_HARMONICNOTCH_ENABLED` in defaults header file instead

Reply:
changed